src/include/dhandle.h

```diff
@@ -112,4 +112,18 @@ struct __wt_data_handle {
 #define WT_DHANDLE_OPEN 0x80u         /* Handle is open */
                                       /* AUTOMATIC FLAG VALUE GENERATION STOP */
     uint32_t flags;
+
```

Comment:
DHANDLE flags are shared by both DHANDLE and BTREE. So increasing those flags can lead to wrong behaviour. So I created a new flags field to store them. As I created a new flag to store these options, I thought of removing the `assert_flags` is proper now.

Reply:
The change makes sense (nice catch), but the comments and formatting needs to be updated.